src/App.tsx

```diff
@@ -1,10 +1,42 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { USER_ID, filterByStatus } from './api/todos';
+import { typeTodo } from './types/Todo';
+import { Todo } from './components/Todo/Todo';
+import { Loader } from './components/Loader/Loader';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterType, setFilterType] = useState('all');
+  const [todos, setTodos] = useState<typeTodo[]>([]);
+
+  const filterStatus = (filter: string) => {
+    let status: string | boolean = '';
+
+    setFilterType(filter);
+
+    if (filter === 'all') {
+      status = 'all';
+    } else if (filter === 'active') {
+      status = false;
+    } else {
+      status = true;
+    }
+
+    setIsLoading(true);
+    filterByStatus(status)
```

Comment:
no need to fetch all todos every time you set filter
Fetch all todos once and save them in state. Then create a common const variable with filtered todos. This variable will be recalculated every time a component rerenders - it means after you change some state (including `filterStatus`)